tests/php/test-amp-wordpress-embed-handler.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Class AMP_WordPress_Embed_Handler_Test
+ *
+ * @package AMP
+ */
+
+use AmpProject\AmpWP\Tests\Helpers\WithoutBlockPreRendering;
+use AmpProject\AmpWP\Tests\TestCase;
+
+/**
+ * Class AMP_WordPress_Embed_Handler_Test
+ *
+ * @covers AMP_WordPress_Embed_Handler
+ */
+class AMP_WordPress_Embed_Handler_Test extends TestCase {
+
+	use WithoutBlockPreRendering;
+
+	/**
+	 * Get conversion data.
+	 *
+	 * @return array
+	 */
+	public function get_conversion_data() {
+		return [
+			'no_embed'   => [
+				'<p>Hello world.</p>',
+				'<p>Hello world.</p>' . PHP_EOL,
+			],
+			'simple_url' => [
+				'https://make.wordpress.org/core/2021/10/12/proposal-for-a-performance-team/' . PHP_EOL,
+				'<p><amp-wordpress-embed height="200" title="&#8220;Proposal for a Performance team&#8221; &#8212; Make WordPress Core" data-url="https://make.wordpress.org/core/2021/10/12/proposal-for-a-performance-team/"></p>' . PHP_EOL .
+				'<blockquote class="wp-embedded-content" data-secret="%s" placeholder><p><a href="https://make.wordpress.org/core/2021/10/12/proposal-for-a-performance-team/">Proposal for a Performance team</a></p></blockquote>' . PHP_EOL .
+				'<p><button overflow type="button">Expand</button></amp-wordpress-embed></p>' . PHP_EOL,
```

Comment:
This doesn't look right. With some formatting, it's clear that `wpautop` is seriously messing this up:

```html
<p>
	<amp-wordpress-embed height="200" title="&#8220;Proposal for a Performance team&#8221; &#8212; Make WordPress Core" data-url="https://make.wordpress.org/core/2021/10/12/proposal-for-a-performance-team/">
</p>
<blockquote class="wp-embedded-content" data-secret="%s" placeholder>
	<p>
		<a href="https://make.wordpress.org/core/2021/10/12/proposal-for-a-performance-team/">Proposal for a Performance team</a>
	</p>
</blockquote>
<p>
	<button overflow type="button">Expand</button>
	</amp-wordpress-embed>
</p>
```

Because of this, we may need to take an alternative approach. Instead of filtering `embed_oembed_html`, we may need to implement the `sanitize_raw_embeds` method instead. This will allow us to operate on the DOM after `the_content` filters have applied, instead of operating on a HTML string which will get corrupted by `wpautop`. I understand the key issue with `wpautop` is the `blockquote`, that it is (rightly) avoiding wrapping a `blockquote` in a `p`, but the problem is that it shouldn't have tried to wrap `amp-wordpress-embed` in the first place.